garage/tf/algos/dqn.py

```diff
@@ -0,0 +1,282 @@
+
+        # Baselines atari wrappers use a LazyFrame object to store the
+        # observation. The way to unpack it is to unpack obs as numpy array.
+        # For reference, see
+        # https://github.com/openai/baselines/blob/1259f6ab25c6f7261e33c
+        # 4c3b92df869188f9260/baselines/common/atari_wrappers.py#L202
+        if self.use_atari_wrappers:
+            obs = np.asarray(obs)
+        episode_rewards.append(0.)
+
+        for itr in range(self.num_timesteps):
+            with logger.prefix('Iteration #%d | ' % itr):
+                # normalize observation for pixel environment
+                if len(self.env.spec.observation_space.shape) == 3:
+                    obs_scaled = obs.astype(np.float32) / 255.0
+                else:
+                    obs_scaled = obs
+
+                if self.es:
+                    action, _ = self.es.get_action(itr, obs_scaled,
+                                                   self.policy)
+                else:
+                    action, _ = self.policy.get_action(obs_scaled)
+
+                next_obs, reward, done, env_info = self.env.step(action)
+
+                if self.use_atari_wrappers:
+                    next_obs = np.asarray(next_obs)
+                self.replay_buffer.add_transitions(
+                    observation=[obs],
+                    action=[action],
+                    reward=[reward],
+                    terminal=[done],
+                    next_observation=[next_obs],
+                )
+
+                episode_rewards[-1] += reward
+                ts += 1
+                obs = next_obs
+
+                if done:
+                    episode_length.append(ts)
+                    ts = 0
+                    episode_rewards.append(0.)
+                    obs = self.env.reset()
+                    if self.use_atari_wrappers:
+                        obs = np.asarray(obs)
+
+                if self.replay_buffer.n_transitions_stored >= self.min_buffer_size:  # noqa: E501
+                    self.evaluate = True
+                    qf_loss = self.optimize_policy(itr, None)
+                    episode_qf_losses.append(qf_loss)
+                    if itr % self.target_network_update_freq == 0:
+                        self.sess.run(self._qf_update_ops, feed_dict=dict())
+
+                if self.plot:
+                    self.plotter.update_plot(self.policy, self.max_path_length)
+                    if self.pause_for_plot:
+                        input("Plotting evaluation run: Press Enter to "
+                              "continue...")
+
+                if self.evaluate:
+                    mean100ep_rewards = round(
+                        np.mean(episode_rewards[-100:]), 1)
+                    mean100ep_qf_loss = np.mean(episode_qf_losses[-100:])
+                    if itr % self.print_freq == 0:
+                        logger.record_tabular('Iteration', itr)
+                        logger.record_tabular('AverageReturn',
```

Comment:
I think it would be better to rename this to Episode100RewardMean. Because in garage convention, AverageReturn means the mean of the current rollout's return.